engine/table/src/main/java/io/deephaven/engine/util/PythonDeephavenSession.java

```diff
@@ -186,6 +198,65 @@ protected void evaluate(String command, String scriptName) {
         return Collections.unmodifiableMap(scope.getEntriesMap());
     }
 
+    private static class PythonSnapshot implements Snapshot, Closeable {
+
+        private final PyDictWrapper dict;
+
+        public PythonSnapshot(PyDictWrapper dict) {
```

Comment:
The old version was very inefficient (getVariables is very inefficient for python). In general, I think our ScriptSession interface as a top-level programmatic object leaves a lot to be desired, esp wrt python. Pushing some of the compute changes logic down into python allows us to avoid crossing the JNI boundary as much. I'd like to take it further at some point, as I think more logic can/should be moved into python.

Reply:
Okay, I'll let this be deferred, but while I also dislike the current AbstractScriptSession/QueryScope abstraction/relationship, I dont think we're tearing out enough in this patch, or designing (i.e. with review) the new approach/api being put here. 

If it is a good idea to remove, I'd rather do it in a complete way.